src/lib.rs

```diff
@@ -173,36 +297,27 @@ use SizeClass::*;
 // TODO: Find a way to test that this actually does what it claims; that is,
 // that it boxes large values and copies small ones.
 #[repr(transparent)]
-pub struct Stowaway<T> {
-    // TODO: Reimplemnt this as union, once we can have non-copy fields in a
+pub struct Stowaway<T: Stowable> {
+    // TODO: Reimplement this as union, once we can have non-copy fields in a
     // union.
     storage: *mut T,
 }
 
-impl<T> Stowaway<T> {
-    /// If true, we're packing a T into a *mut T. Otherwise, we're
-    /// using a box.
-    ///
+impl<T: Stowable> Stowaway<T> {
     /// TODO: make this a const fn when `if` is allowed in const
     #[inline(always)]
     fn size_class() -> SizeClass {
-        assert_eq!(
-            mem::size_of::<*mut T>(),
-            mem::size_of::<*mut ()>(),
-            "Cannot currently stow pointers to DSTs"
-        );
-
-        if mem::align_of::<T>() > mem::align_of::<*mut T>() {
-            // Need to unconditionally box values that don't match our
-            // alignment, even if sizeof(T) <= sizeof(Self)
-            Boxed
-        } else if mem::size_of::<T>() == 0 {
+        if mem::size_of::<T>() == 0 {
             Zero
         } else if mem::size_of::<T>() <= mem::size_of::<*mut T>() {
             Packed
         } else {
             Boxed
         }
+        // Previously we checked the alignment of T, but the only way for
+        // alignof(T) > alignof(ptr) to be true is if
+        // sizeof(t) == 0 or sizeof(T) > sizeof(ptr), both of which are
+        // handled.
```

Comment:
Is this documented somewhere? I not can you add this check because there shouldn't be any cost to doing so, const propogation should eliminate the check.

Reply:
Yup:
https://stackoverflow.com/questions/61256701/is-it-possible-to-have-a-type-with-a-larger-alignment-than-its-own-size
https://doc.rust-lang.org/reference/type-layout.html#size-and-alignment
> The size of a value is always a multiple of its alignment.

Reply:
I'm happy to restore it, but we require this property to be true in order for `[T; N]` to be stowable when `T: Stowable`, so I was happy to assume it here.

Reply:
Ok, that works for me